### PR TITLE
Refactor Files class into proper classes

### PR DESCRIPTION
**Pull request by pem70**
**What It Does**
Refactor files.py into separate files. Class Files will be supported still, and each module can run independently.

**How to Test**

**Review Checklist**
I certify that I have:
- [x] tested my changes
- [x] added/updated automated tests
- [x] updated the changelog
- [x] followed the [contribution guidelines](https://github.com/zowe/zowe-cli/blob/master/CONTRIBUTING.md)

**Additional Comments**

### CODE REVIEW

File: src/zos_files/zowe/zos_files_for_zowe_sdk/files.py [hunk 1]
```diff
@@ -31,10 +34,14 @@ class Files(SdkApi):
     connection
         connection object
     """
+    dsn: Datasets
```
Comment (traeok) [inline src/zos_files/zowe/zos_files_for_zowe_sdk/files.py:37]: I'm wondering if we want to rename this variable to `ds` to be consistent with the API endpoint? (which is  `/zosmf/restfiles/ds` for data sets)

**Review by traeok — APPROVED** [review]
LGTM @pem70 thanks for splitting these up into classes and updating the deprecated functions! 👍

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,7 +6,7 @@ All notable changes to the Zowe Client Python SDK will be documented in this fil
 
 ### Enhancements
 
-- Added logger class to core SDK [#185](https://github.com/zowe/zowe-client-python-sdk/issues/185)
+- Refactor Files class into proper classes [#264](https://github.com/zowe/zowe-client-python-sdk/issues/264)
```
Comment (zFernand0) [inline CHANGELOG.md:9]: Should we list the new classes (e.g. Datasets, USSFiles, FileSystems) as the enhancements?

Maybe something like:
```suggestion
- Refactored Files class into `Datasets`, `USSFiles`, and `FileSystems`. [#264](https://github.com/zowe/zowe-client-python-sdk/issues/264)
```

Or something like this:
```suggestion
- Added classes for handling `Datasets`, `USSFiles`, and `FileSystems` in favor of the single Files class. [#264](https://github.com/zowe/zowe-client-python-sdk/issues/264)
```

File: src/zos_files/zowe/zos_files_for_zowe_sdk/files.py (hunk 1 above)
Comment (zFernand0) [reply src/zos_files/zowe/zos_files_for_zowe_sdk/files.py:37]: I think the suggestion to change `dsn` with `ds` across the board makes sense for consistency.

**Review by zFernand0 — APPROVED** [review]
Changes LGTM! 😋 
I like the direction that was taken with the refactoring.
Left two minor comments, both of which can be addressed in a new PR that combines all 3 issues:

- https://github.com/zowe/zowe-client-python-sdk/issues/264
- https://github.com/zowe/zowe-client-python-sdk/issues/265
- https://github.com/zowe/zowe-client-python-sdk/issues/104

**Comment by zFernand0** [comment]
Closing in favor of the next PR in line:
- #275

**Pull request closed without merging**